### PR TITLE
[RSDK-5462] More gps cleanup

### PR DESCRIPTION
**Pull request by penguinland**
No behavioral changes are intended. This is a follow-up to https://github.com/viamrobotics/rdk/pull/3548: I removed the duplicated code, but didn't notice that the control flow within that code was so messy. 

Before this PR, when we have successfully connected, `err` would be `nil`, so we would break out of the for loop. We could get out of the for loop either by successfully connecting or by failing to connect too many times, so we'd then check if the error is not `nil` to find out which situation we're in, and then store the client and return. After this PR, when we successfully connect, we just store the client and return, and if we ever get out of the for loop, it's because we were unable to connect after the maximum number of attempts.

My next instinct was to make `c` and `err` local within the for loop, and stop declaring them before their first use. However, this makes it difficult to return the most recent error, if you get past the for loop. So, I added in a comment about that (but don't really like the way it's phrased; if you've got a better suggestion, I'm all for it!).

### CODE REVIEW

File: components/movementsensor/rtkutils/ntrip.go [hunk 1]
```diff
@@ -223,6 +223,8 @@ func parseStream(line string) (Stream, error) {
 
 // Connect attempts to initialize a new ntrip client.
 func (n *NtripInfo) Connect(ctx context.Context, logger logging.Logger) error {
+	// If we're unable to connect after multiple attempts, we return the last error. Declare the
+	// error variable out here so its lifetime is more than just the for loop.
```
Comment (JohnN193) [inline components/movementsensor/rtkutils/ntrip.go:227]: Wording I think is fine, I just have two thoughts here.

1. should the first sentence be a part of the description of the function?
2. Not sure if the second sentence is necessary, but there’s no reason to remove the comment if ya like it

File: components/movementsensor/rtkutils/ntrip.go [hunk 2]
```diff
@@ -235,19 +237,15 @@ func (n *NtripInfo) Connect(ctx context.Context, logger logging.Logger) error {
 		}
 
 		c, err = ntrip.NewClient(n.URL, ntrip.Options{Username: n.username, Password: n.password})
```
Comment (JohnN193) [inline components/movementsensor/rtkutils/ntrip.go:239]: If you want to keep the second sentence, maybe add a comment here to mention why we do not declare the client here as well?

**Review by JohnN193 — APPROVED** [review]
LGTM, I think the comments are fine as is, though maybe they can be moved around/added to depending on what you want a reader to get out of them

File: components/movementsensor/rtkutils/ntrip.go (hunk 1 above)
Comment (penguinland) [reply components/movementsensor/rtkutils/ntrip.go:227]: Oh dang, point 1 is a really good one! Thanks. and I like your other suggestion of moving the other half of this comment further down.

Comment (penguinland) [reply components/movementsensor/rtkutils/ntrip.go:227]: ...and you're even more right. After moving the first sentence up, there is no good place for the second sentence at all, and the function description primes you to understand why the variable declarations are so early, so the second sentence can be removed. Thanks so much!

**Pull request merged**